src/Nethermind/Nethermind.Core/Buffers/ISpanSource.cs

```diff
@@ -0,0 +1,37 @@
+// SPDX-FileCopyrightText: 2025 Demerzel Solutions Limited
+// SPDX-License-Identifier: LGPL-3.0-only
+
+using System;
+
+namespace Nethermind.Core.Buffers;
+
+/// <summary>
+/// A tiny array, allocating just enough space to hold the payload with a byte-long length.
+/// </summary>
+public interface ISpanSource
+{
+    /// <summary>
+    /// The length of the underlying payload.
+    /// </summary>
+    int Length { get; }
+
+    /// <summary>
+    /// Like <see cref="MemoryExtensions.SequenceEqual{T}(System.ReadOnlySpan{T},System.ReadOnlySpan{T})"/>
+    /// </summary>
+    bool SequenceEqual(ReadOnlySpan<byte> other);
+
+    /// <summary>
+    /// Like <see cref="MemoryExtensions.CommonPrefixLength{T}(System.ReadOnlySpan{T},System.ReadOnlySpan{T})"/>
+    /// </summary>
+    int CommonPrefixLength(ReadOnlySpan<byte> other);
```

Comment:
Do we need those when we have `Span`? Maybe they should be extension methods?

Reply:
The idea was to have them as a special case of `TinyArray` can provide a better performance (unrolled vectorized comparison for example). I think I'll have them removed as well as the `TinyArray` implementations.